crates/ide/src/hover/render.rs

```diff
@@ -715,10 +715,17 @@ pub(super) fn literal(
     let mut s = format!("```rust\n{ty}\n```\n___\n\n");
     match value {
         Ok(value) => {
+            let backtick_len = value.chars().filter(|c| *c == '`').count();
+            let backticks = "`".repeat(backtick_len + 1);
+
             if let Some(newline) = value.find('\n') {
-                format_to!(s, "value of literal (truncated up to newline): {}", &value[..newline])
+                format_to!(
+                    s,
+                    "value of literal (truncated up to newline): {backticks} {} {backticks}",
+                    &value[..newline]
+                )
             } else {
-                format_to!(s, "value of literal: {value}")
+                format_to!(s, "value of literal: {backticks} {value} {backticks}")
```

Comment:
```suggestion
                format_to!(s, "value of literal: {backticks}{value}{backticks}")
```

Reply:
Ah wait I just realized this is allowed by the spec (and required to quote a backtick) nvm

Reply:
I forgot to mention in the PR initial comment, the space is needed, otherwise single back tick or back tick at last position will mess the formatting